### PR TITLE
Euen.failing branch

### PR DESCRIPTION
**Pull request by Euen**

### CODE REVIEW

File: src/erlang_repo_sup.erl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+-module(erlang_repo_sup).
+-behaviour(supervisor).
+
+-export([start_link/0, init/1]).
+
+start_link() ->
+    supervisor:start_link({local, ?MODULE}, ?MODULE, []).
+
+init([]) ->
+	Procs = [],
```
Comment (gadgetci) [inline src/erlang_repo_sup.erl]: According to **Elvis**:

> Line 10 has a tab at column 0.

File: src/erlang_repo_sup.erl [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+	{ok, {{one_for_one, 1, 5}, Procs}}.
```
Comment (gadgetci) [inline src/erlang_repo_sup.erl]: According to **Elvis**:

> Line 11 has a tab at column 0.

**Comment by gadgetci** [comment]
**dialyzer** failed with exit status: 2

File: src/erlang_repo_sup.erl [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+start_link() ->
+    supervisor:start_link({local, ?MODULE}, ?MODULE, []).
+
+init([]) ->
+    Procs = [],
+    erlang_repo_sup:fin(),
+    {ok, {{one_for_one, 1, 5}, Procs}}.
+
+fin() -> 1 + 1.
```
Comment (gadgetci) [inline src/erlang_repo_sup.erl]: According to **Dialyzer**:

> function fin/0 is unused

File: src/erlang_repo_sup.erl [hunk 4]
```diff
@@ -0,0 +1,12 @@
+-module(erlang_repo_sup).
+-behaviour(supervisor).
+
+-export([start_link/0, init/1]).
+
+start_link() ->
+    supervisor:start_link({local, ?MODULE}, ?MODULE, []).
+
+-spec init(boolean()) -> boolean().
```
Comment (gadget-tester) [inline src/erlang_repo_sup.erl]: According to **Dialyzer**:

>  The return type boolean() in the specification of init/1 is not a subtype of 'ignore' | {'ok',{{'one_for_all',non_neg_integer(),pos_integer()} | {'one_for_one',non_neg_integer(),pos_integer()} | {'rest_for_one',non_neg_integer(),pos_integer()} | {'simple_one_for_one',non_neg_integer(),pos_integer()} | #{},[{_,{atom() | tuple(),atom(),'undefined' | [any()]},'permanent' | 'temporary' | 'transient','brutal_kill' | 'infinity' | non_neg_integer(),'supervisor' | 'worker','dynamic' | [atom() | tuple()]} | #{}]}}, which is the expected return type for the callback of supervisor behaviour

Comment (gadget-tester) [inline src/erlang_repo_sup.erl]: According to **Dialyzer**:

>  Invalid type specification for function erlang_repo_sup:init/1. The success typing is ([]) -> {'ok',{{'one_for_one',1,5},[]}}

File: src/erlang_repo_sup.erl [hunk 5]
```diff
@@ -0,0 +1,11 @@
+-module(erlang_repo_sup).
+-behaviour(supervisor).
+
+-export([start_link/0, init/1, end/0]).
+
+start_link() ->
```
Comment (gadget-tester) [inline src/erlang_repo_sup.erl:6]: According to **Dialyzer**:

> function start_link/0 is unused

File: src/erlang_repo_sup.erl [hunk 6]
```diff
@@ -0,0 +1,11 @@
+-module(erlang_repo_sup).
+-behaviour(supervisor).
```
Comment (gadget-tester) [inline src/erlang_repo_sup.erl:2]: According to **Dialyzer**:

> undefined callback function init/1 (behaviour 'supervisor')

File: src/erlang_repo_sup.erl [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+-export([start_link/0, init/1, end/0]).
```
Comment (gadget-tester) [inline src/erlang_repo_sup.erl]: According to **Dialyzer**:

> syntax error before: 'end'

File: src/erlang_repo_sup.erl [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+start_link() ->
+    supervisor:start_link({local, ?MODULE}, ?MODULE, []).
+
+init([]) ->
```
Comment (gadget-tester) [inline src/erlang_repo_sup.erl:9]: According to **Dialyzer**:

> function init/1 is unused

**Pull request closed without merging**